src/lib/address/address.ts

```diff
@@ -151,7 +155,23 @@ export type ParseAddressInputReturn = {
 }
 export type ParseAddressInput = (args: ParseAddressInputArgs) => Promise<ParseAddressInputReturn>
 
-export const parseAddressInput: ParseAddressInput = async args => {
+export const parseAddressInput: ParseAddressInput = async ({ value, chainId }) => {
+  let eip681Address
+
+  try {
+    eip681Address = parse(value)
+  } catch (error) {
+    moduleLogger.trace(error, 'cannot parse eip681 address')
+  }
+
+  const args: ParseAddressInputArgs = {
+    value:
+      eip681Address?.target_address && !eip681Address.parameters
+        ? eip681Address.target_address
+        : value,
+    chainId,
+  }
+
```

Comment:
This isn't discriminating by `chainId`, which means this branch will be reached even when parsing UTXO addresses / URLs:

<img width="545" alt="image" src="https://user-images.githubusercontent.com/17035424/191487073-a25598bc-1798-483e-baa3-64d7f947b699.png">

It won't be an issue at runtime as we will properly validate addresses by chainId, but this brings two issues at an architectural level:

1. We should never reach a branch with tight coupling to a specific chain/Id, when in the context of another chain/Id
2. As a result of 1., we will be unable to add parsing for other chain/Ids, e.g BIP-21 that we will also need to support (EIP-681 parsing does not work with it, as is intended as these are two very similar standards, but related to two different chains with their own chain-specific ramifications)

I will not block the PR for that, but going forward, we should implement a mapping of chainId to URL/parser similar to what `validateVanityAddress` is doing:

https://github.com/shapeshift/web/blob/26649710866034cd3679973e1f08358961cc5e43/src/lib/address/address.ts#L19-L21

- For BIP-21/EIP-681 and possible other URL formats we could be supporting, it would be `(value: UrlOrAddress): Address`
- For addresses, it will be a simple [`identity`](https://lodash.com/docs/4.17.15#identity) function: `(value: Address) => value`